tests/unit/contentproviders/test_ckan.py

```diff
@@ -0,0 +1,58 @@
+import os
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile, TemporaryDirectory
+
+import pytest
+
+from repo2docker.contentproviders import CKAN
+
+test_ckan = CKAN()
+test_hosts = [
+    (
+        [
+            "http://demo.ckan.org/dataset/sample-dataset-1",
+        ],
+        {
+            "dataset_id": "sample-dataset-1",
+            "api_url": "http://demo.ckan.org/api/3/action/",
+            "version": "1707387710",
+        },
+    )
+]
+
+
+@pytest.mark.parametrize("test_input, expected", test_hosts)
+def test_detect_ckan(test_input, expected):
+    assert CKAN().detect(test_input[0]) == expected
+
+    # Don't trigger the CKAN content provider
+    assert CKAN().detect("/some/path/here") is None
```

Comment:
Since these are not using the parameterized information, these should be their own test.

Reply:
Yes, they should. Fixed.